From: Baoquan He <bhe@redhat.com>
Subject: mm/gup: fix the outdated code comments above get_user_pages_unlocked()
Date: Mon, 31 Mar 2025 16:13:23 +0800

Since commit f0818f472d8d ("mm: gup: add get_user_pages_locked and
get_user_pages_unlocked"), get_user_pages() doesn't need to have mmap_lock
held anymore.  It calls __get_user_pages_locked() which can acquire and
drop the mmap_lock internaly.

Hence remove the incorrect code comments now.

Link: https://lkml.kernel.org/r/20250331081327.256412-4-bhe@redhat.com
Signed-off-by: Baoquan He <bhe@redhat.com>
Cc: <x86@kernel.org>
Cc: Yanjun.Zhu <yanjun.zhu@linux.dev>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/gup.c |   16 +++-------------
 1 file changed, 3 insertions(+), 13 deletions(-)

--- a/mm/gup.c~mm-gup-fix-the-outdated-code-comments-above-get_user_pages_unlocked
+++ a/mm/gup.c
@@ -2702,19 +2702,9 @@ long get_user_pages(unsigned long start,
 EXPORT_SYMBOL(get_user_pages);
 
 /*
- * get_user_pages_unlocked() is suitable to replace the form:
- *
- *      mmap_read_lock(mm);
- *      get_user_pages(mm, ..., pages, NULL);
- *      mmap_read_unlock(mm);
- *
- *  with:
- *
- *      get_user_pages_unlocked(mm, ..., pages);
- *
- * It is functionally equivalent to get_user_pages_fast so
- * get_user_pages_fast should be used instead if specific gup_flags
- * (e.g. FOLL_FORCE) are not required.
+ * get_user_pages_unlocked() is functionally equivalent to
+ * get_user_pages_fast so get_user_pages_fast should be used instead
+ * if specific gup_flags (e.g. FOLL_FORCE) are not required.
  */
 long get_user_pages_unlocked(unsigned long start, unsigned long nr_pages,
 			     struct page **pages, unsigned int gup_flags)
_
